thrift/shared.thrift

```diff
@@ -1734,6 +1734,15 @@ enum CrossClusterTaskFailedCause {
   DOMAIN_NOT_EXISTS
   WORKFLOW_ALREADY_RUNNING
   WORKFLOW_NOT_EXISTS
+  WORKFLOW_ALREADY_COMPLETED
+  UNCATEGORIZED
+}
+
+enum GetCrossClusterTaskFailedCause {
```

Comment:
@yux0 Do you think replication stack may reuse this type? I can rename it to `GetTaskFailedCause` if so.

Reply:
Yes. I think this is a good idea. Replication can use this in the future

Reply:
Sounds good. Renamed to `GetTaskFailedCause`